diff --git a/config/win/BUILD.gn b/config/win/BUILD.gn
--- a/config/win/BUILD.gn
+++ b/config/win/BUILD.gn
@@ -305,7 +305,7 @@
 # manually override this config for their compiles.
 config("winver") {
   defines = [
-    "NTDDI_VERSION=NTDDI_WIN11_BR",
+    "NTDDI_VERSION=NTDDI_WIN11_GE",

     # We can't say `=_WIN32_WINNT_WIN10` here because some files do
     # `#if WINVER < 0x0600` without including windows.h before,
diff --git a/toolchain/win/setup_toolchain.py b/toolchain/win/setup_toolchain.py
index e7651ec9f..bce79d530 100644
--- a/toolchain/win/setup_toolchain.py
+++ b/toolchain/win/setup_toolchain.py
@@ -22,7 +22,7 @@ import action_helpers
 import gn_helpers
 
 SCRIPT_DIR = os.path.dirname(__file__)
-SDK_VERSION = '10.0.28000.0'
+SDK_VERSION = '10.0.26100.0'
 MSVC_DIR = re.compile('^.*/VC/Tools/MSVC/[^/]+/include$')
 WINDOWS_KITS_DIR = re.compile(r'^(.*/Windows Kits/\d+/Include/[^/]+)/.*')
 
diff --git a/vs_toolchain.py b/vs_toolchain.py
index 8bcba3e3f..048db2b15 100755
--- a/vs_toolchain.py
+++ b/vs_toolchain.py
@@ -60,7 +60,7 @@ from gn_helpers import ToGNString
 #   Make sure any version numbers in the documentation match the code.
 #
 TOOLCHAIN_HASH = '3bfcb536c8'
-SDK_VERSION = '10.0.28000.0'
+SDK_VERSION = '10.0.26100.0'
 
 # Visual Studio versions are listed in descending order of priority.
 # The first version is assumed by this script to be the one that is packaged,
