--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -244,8 +244,10 @@
 elseif(MSVC)
   target_link_libraries(hermesvm PRIVATE hermesvm_a)
   target_link_options(hermesvm PRIVATE /WHOLEARCHIVE:hermesvm_a)
+  target_link_options(hermesvm PRIVATE "/WHOLEARCHIVE:$<TARGET_FILE:jsi>")
   target_link_libraries(hermesvmlean PRIVATE hermesvmlean_a)
   target_link_options(hermesvmlean PRIVATE /WHOLEARCHIVE:hermesvmlean_a)
+  target_link_options(hermesvmlean PRIVATE "/WHOLEARCHIVE:$<TARGET_FILE:jsi>")
 else()
   target_link_libraries(hermesvm PRIVATE
     -Wl,--whole-archive
