diff --git a/lib/VM/Operations.cpp b/lib/VM/Operations.cpp
--- a/lib/VM/Operations.cpp
+++ b/lib/VM/Operations.cpp
@@ -3215,74 +3215,64 @@
 }
 } // namespace
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_sub_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_sub_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
   return binOpImpl<doSub, BigIntPrimitive::subtract>(shr, a, b);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_mul_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_mul_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
   return binOpImpl<doMul, BigIntPrimitive::multiply>(shr, a, b);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_div_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_div_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
   return binOpImpl<doDiv, BigIntPrimitive::divide>(shr, a, b);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_mod_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_mod_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
   return binOpImpl<doMod, BigIntPrimitive::remainder>(shr, a, b);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_inc_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_inc_rjs(
     SHRuntime *shr,
     const SHLegacyValue *n) {
   return incDecOperImpl<doInc, BigIntPrimitive::inc>(shr, n);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_dec_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_dec_rjs(
     SHRuntime *shr,
     const SHLegacyValue *n) {
   return incDecOperImpl<doDec, BigIntPrimitive::dec>(shr, n);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_bit_and_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_bit_and_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
   return bitOperImpl<doBitAnd, BigIntPrimitive::bitwiseAND>(shr, a, b);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_bit_or_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_bit_or_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
   return bitOperImpl<doBitOr, BigIntPrimitive::bitwiseOR>(shr, a, b);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_bit_xor_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_bit_xor_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
   return bitOperImpl<doBitXor, BigIntPrimitive::bitwiseXOR>(shr, a, b);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_left_shift_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_left_shift_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
@@ -3290,8 +3280,7 @@
       shr, a, b);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_unsigned_right_shift_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_unsigned_right_shift_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
@@ -3301,8 +3290,7 @@
       BigIntPrimitive::unsignedRightShift>(shr, a, b);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_right_shift_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_right_shift_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
@@ -3312,8 +3300,7 @@
       BigIntPrimitive::signedRightShift>(shr, a, b);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_bit_not_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_bit_not_rjs(
     SHRuntime *shr,
     const SHLegacyValue *n) {
   Handle<> nHandle{toPHV(n)};
@@ -3342,8 +3329,7 @@
   return *res;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_minus_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_minus_rjs(
     SHRuntime *shr,
     const SHLegacyValue *n) {
   // Fast path, n is a number.
@@ -3375,8 +3361,7 @@
   return toBoolean(HermesValue::fromRaw(b.raw));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" bool _sh_ljs_equal_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE bool _sh_ljs_equal_rjs(
     SHRuntime *shr,
     const SHLegacyValue *a,
     const SHLegacyValue *b) {
@@ -3396,13 +3381,11 @@
       HermesValue::fromRaw(a.raw), HermesValue::fromRaw(b.raw));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_typeof(SHRuntime *shr, SHLegacyValue *v) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_typeof(SHRuntime *shr, SHLegacyValue *v) {
   return typeOf(getRuntime(shr), Handle<>::vmcast(toPHV(v)));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" bool _sh_ljs_typeof_is(SHLegacyValue val, uint16_t typesRaw) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE bool _sh_ljs_typeof_is(SHLegacyValue val, uint16_t typesRaw) {
   hermes::TypeOfIsTypes types(typesRaw);
   HermesValue hv = HermesValue::fromRaw(val.raw);
   return matchTypeOfIs(hv, types);
@@ -3435,8 +3418,7 @@
   }
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_instance_of_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_instance_of_rjs(
     SHRuntime *shr,
     SHLegacyValue *object,
     SHLegacyValue *constructor) {
@@ -3452,8 +3434,7 @@
   return _sh_ljs_bool(*cr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_iterator_begin_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_iterator_begin_rjs(
     SHRuntime *shr,
     SHLegacyValue *src) {
   auto res = [&]() -> CallResult<HermesValue> {
@@ -3495,8 +3476,7 @@
   return *res;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_iterator_next_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_iterator_next_rjs(
     SHRuntime *shr,
     SHLegacyValue *iteratorOrIdx,
     const SHLegacyValue *srcOrNext) {
@@ -3591,8 +3571,7 @@
   return *res;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_iterator_close_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_iterator_close_rjs(
     SHRuntime *shr,
     const SHLegacyValue *iteratorOrIdx,
     bool ignoreExceptions) {
diff --git a/lib/VM/StaticH.cpp b/lib/VM/StaticH.cpp
--- a/lib/VM/StaticH.cpp
+++ b/lib/VM/StaticH.cpp
@@ -55,8 +55,7 @@
 }
 } // namespace
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw_register_stack_overflow(SHRuntime *shr) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw_register_stack_overflow(SHRuntime *shr) {
   Runtime &runtime = getRuntime(shr);
   (void)runtime.raiseStackOverflow(Runtime::StackOverflowKind::JSRegisterStack);
   _sh_throw_current(shr);
@@ -71,8 +70,7 @@
   }
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_coerce_this_ns(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_coerce_this_ns(
     SHRuntime *shr,
     SHLegacyValue value) {
   if (LLVM_LIKELY(_sh_ljs_is_object(value))) {
@@ -126,24 +124,21 @@
   return res->getHermesValue();
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_arguments_prop_by_val_loose(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_arguments_prop_by_val_loose(
     SHRuntime *shr,
     SHLegacyValue *frame,
     SHLegacyValue *idx,
     SHLegacyValue *lazyReg) {
   return getArgumentsPropByVal_RJS(shr, frame, idx, lazyReg, false);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_arguments_prop_by_val_strict(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_arguments_prop_by_val_strict(
     SHRuntime *shr,
     SHLegacyValue *frame,
     SHLegacyValue *idx,
     SHLegacyValue *lazyReg) {
   return getArgumentsPropByVal_RJS(shr, frame, idx, lazyReg, true);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_arguments_length(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_arguments_length(
     SHRuntime *shr,
     SHLegacyValue *frame,
     SHLegacyValue *lazyReg) {
@@ -194,23 +189,20 @@
   *lazyReg = *res;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_reify_arguments_loose(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_reify_arguments_loose(
     SHRuntime *shr,
     SHLegacyValue *frame,
     SHLegacyValue *lazyReg) {
   reifyArguments(shr, frame, lazyReg, false);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_reify_arguments_strict(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_reify_arguments_strict(
     SHRuntime *shr,
     SHLegacyValue *frame,
     SHLegacyValue *lazyReg) {
   reifyArguments(shr, frame, lazyReg, true);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_by_val_with_receiver_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_by_val_with_receiver_rjs(
     SHRuntime *shr,
     SHLegacyValue *source,
     SHLegacyValue *key,
@@ -258,8 +250,7 @@
   return res->getHermesValue();
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_ljs_get_by_index_rjs(SHRuntime *shr, SHLegacyValue *source, uint32_t key) {
   Handle<> sourceHandle{toPHV(source)};
   Runtime &runtime = getRuntime(shr);
@@ -288,8 +279,7 @@
   return res->getHermesValue();
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_own_private_by_sym(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_own_private_by_sym(
     SHRuntime *shr,
     const SHLegacyValue *source,
     const SHLegacyValue *privateNameKey,
@@ -326,8 +316,7 @@
   return *res;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_catch(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_catch(
     SHRuntime *shr,
     SHLocals *locals,
     SHLegacyValue *frame,
@@ -338,8 +327,7 @@
   return _sh_get_clear_thrown_value(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_catch_no_pop(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_catch_no_pop(
     SHRuntime *shr,
     SHLocals *locals,
     SHLegacyValue *frame,
@@ -351,16 +339,14 @@
   runtime.setCurrentFrame(StackFramePtr(toPHV(frame)));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_get_clear_thrown_value(SHRuntime *shr) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_get_clear_thrown_value(SHRuntime *shr) {
   Runtime &runtime = getRuntime(shr);
   SHLegacyValue res = runtime.getThrownValue();
   runtime.clearThrownValue();
   return res;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw_current(SHRuntime *shr) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw_current(SHRuntime *shr) {
   Runtime &runtime = getRuntime(shr);
   assert(runtime.shCurJmpBuf && "No SH exception handler installed");
   if (!runtime.shCurJmpBuf) {
@@ -373,29 +359,25 @@
   llvm_unreachable("longjmp cannot return");
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw(SHRuntime *shr, SHLegacyValue value) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw(SHRuntime *shr, SHLegacyValue value) {
   Runtime &runtime = getRuntime(shr);
   runtime.setThrownValue(HermesValue::fromRaw(value.raw));
   _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw_type_error(SHRuntime *shr, SHLegacyValue *message) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw_type_error(SHRuntime *shr, SHLegacyValue *message) {
   (void)getRuntime(shr).raiseTypeError(Handle<>(toPHV(message)));
   _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw_type_error_ascii(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw_type_error_ascii(
     SHRuntime *shr,
     const char *message) {
   (void)getRuntime(shr).raiseTypeError(TwineChar16(message));
   _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw_reference_error_ascii(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw_reference_error_ascii(
     SHRuntime *shr,
     const char *message) {
   Runtime &runtime = getRuntime(shr);
@@ -403,8 +385,7 @@
   _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw_empty(SHRuntime *shr) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw_empty(SHRuntime *shr) {
   Runtime &runtime = getRuntime(shr);
   (void)runtime.raiseReferenceError("accessing an uninitialized variable");
   _sh_throw_current(shr);
@@ -451,8 +432,7 @@
   return res->getHermesValue();
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_ljs_call(SHRuntime *shr, SHLegacyValue *frame, uint32_t argCount) {
   Runtime &runtime = getRuntime(shr);
   StackFramePtr newFrame(runtime.getStackPointer());
@@ -464,8 +444,7 @@
   return doCall(runtime, &newFrame.getCalleeClosureOrCBRef());
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_callRequire(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_callRequire(
     SHRuntime *shr,
     SHArrayStorage **exportCache,
     SHLegacyValue *requireFunc,
@@ -521,8 +500,7 @@
   return *slowRes;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_call_builtin(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_call_builtin(
     SHRuntime *shr,
     SHLegacyValue *frame,
     uint32_t argCount,
@@ -556,8 +534,7 @@
       getRuntime(shr).getBuiltinCallable(builtinMethodID));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_create_environment(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_create_environment(
     SHRuntime *shr,
     const SHLegacyValue *parentEnv,
     uint32_t size) {
@@ -574,8 +551,7 @@
   // #endif
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_ljs_get_env(SHRuntime *shr, SHLegacyValue startEnv, uint32_t level) {
   Runtime &runtime = getRuntime(shr);
   Environment *curEnv = vmcast<Environment>(*toPHV(&startEnv));
@@ -587,8 +563,7 @@
   return HermesValue::encodeObjectValue(curEnv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_store_to_env(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_store_to_env(
     SHRuntime *shr,
     SHLegacyValue env,
     SHLegacyValue val,
@@ -608,8 +583,7 @@
       .setNonPtr(HermesValue::fromRaw(val.raw), getRuntime(shr).getHeap());
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_create_generator_object(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_create_generator_object(
     SHRuntime *shr,
     const SHLegacyValue *env,
     SHLegacyValue (*func)(SHRuntime *),
@@ -658,8 +632,7 @@
   return *genObjRes;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_create_class(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_create_class(
     SHRuntime *shr,
     const SHLegacyValue *env,
     SHLegacyValue (*func)(SHRuntime *),
@@ -684,8 +657,7 @@
   return HermesValue::encodeObjectValue(std::get<0>(*classRes));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_create_closure(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_create_closure(
     SHRuntime *shr,
     const SHLegacyValue *env,
     SHLegacyValue (*func)(SHRuntime *),
@@ -703,8 +675,7 @@
       .getHermesValue();
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_declare_global_var(SHRuntime *shr, SHSymbolID name) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_declare_global_var(SHRuntime *shr, SHSymbolID name) {
   Runtime &runtime = getRuntime(shr);
   {
     GCScopeMarkerRAII mark{runtime};
@@ -870,8 +841,7 @@
   }
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_put_by_id_loose_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_put_by_id_loose_rjs(
     SHRuntime *shr,
     SHUnit *unit,
     SHLegacyValue *target,
@@ -887,8 +857,7 @@
       reinterpret_cast<WritePropertyCacheEntry *>(propCacheEntry));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_put_by_id_strict_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_put_by_id_strict_rjs(
     SHRuntime *shr,
     SHUnit *unit,
     SHLegacyValue *target,
@@ -904,8 +873,7 @@
       reinterpret_cast<WritePropertyCacheEntry *>(propCacheEntry));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_try_put_by_id_loose_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_try_put_by_id_loose_rjs(
     SHRuntime *shr,
     SHUnit *unit,
     SHLegacyValue *target,
@@ -921,8 +889,7 @@
       reinterpret_cast<WritePropertyCacheEntry *>(propCacheEntry));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_try_put_by_id_strict_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_try_put_by_id_strict_rjs(
     SHRuntime *shr,
     SHUnit *unit,
     SHLegacyValue *target,
@@ -986,24 +953,21 @@
   putByValWithReceiver_RJS(shr, target, key, value, target, strictMode);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_put_by_val_loose_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_put_by_val_loose_rjs(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key,
     SHLegacyValue *value) {
   putByVal_RJS(shr, target, key, value, false);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_put_by_val_strict_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_put_by_val_strict_rjs(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key,
     SHLegacyValue *value) {
   putByVal_RJS(shr, target, key, value, true);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_put_by_val_with_receiver_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_put_by_val_with_receiver_rjs(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key,
@@ -1167,8 +1131,7 @@
       runtime, source, symID, source, cacheEntry);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_create_this(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_create_this(
     SHRuntime *shr,
     SHLegacyValue *callee,
     SHLegacyValue *newTarget,
@@ -1269,8 +1232,7 @@
   _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_try_get_by_id_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_try_get_by_id_rjs(
     SHRuntime *shr,
     const SHLegacyValue *source,
     SHSymbolID symID,
@@ -1281,8 +1243,7 @@
       SymbolID::unsafeCreate(symID),
       reinterpret_cast<ReadPropertyCacheEntry *>(propCacheEntry));
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_by_id_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_by_id_rjs(
     SHRuntime *shr,
     const SHLegacyValue *source,
     SHSymbolID symID,
@@ -1294,8 +1255,7 @@
       reinterpret_cast<ReadPropertyCacheEntry *>(propCacheEntry));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_by_id_with_receiver_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_by_id_with_receiver_rjs(
     SHRuntime *shr,
     const SHLegacyValue *source,
     const SHLegacyValue *receiver,
@@ -1309,8 +1269,7 @@
       reinterpret_cast<ReadPropertyCacheEntry *>(propCacheEntry));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_define_own_by_id(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_define_own_by_id(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHSymbolID key,
@@ -1369,8 +1328,7 @@
     _sh_throw_current(getSHRuntime(runtime));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_define_own_by_val(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_define_own_by_val(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key,
@@ -1390,8 +1348,7 @@
   if (LLVM_UNLIKELY(cr == ExecutionStatus::EXCEPTION))
     _sh_throw_current(shr);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_define_own_ne_by_val(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_define_own_ne_by_val(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key,
@@ -1412,8 +1369,7 @@
     _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_define_own_by_index(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_define_own_by_index(
     SHRuntime *shr,
     SHLegacyValue *target,
     uint32_t key,
@@ -1437,8 +1393,7 @@
 }
 
 /// Put a non-enumerable property.
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_define_own_getter_setter_by_val(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_define_own_getter_setter_by_val(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key,
@@ -1488,8 +1443,7 @@
     _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_add_own_private_by_sym(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_add_own_private_by_sym(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key,
@@ -1507,8 +1461,7 @@
   if (LLVM_UNLIKELY(cr == ExecutionStatus::EXCEPTION))
     _sh_throw_current(shr);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_put_own_private_by_sym(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_put_own_private_by_sym(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *privateNameKey,
@@ -1602,30 +1555,26 @@
   }
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_del_by_val_strict(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_del_by_val_strict(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key) {
   return delByVal(getRuntime(shr), toPHV(target), toPHV(key), true);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_del_by_val_loose(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_del_by_val_loose(
     SHRuntime *shr,
     SHLegacyValue *target,
     SHLegacyValue *key) {
   return delByVal(getRuntime(shr), toPHV(target), toPHV(key), false);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_string(SHRuntime *shr, SHSymbolID symID) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_string(SHRuntime *shr, SHSymbolID symID) {
   NoHandleScope noHandles{getRuntime(shr)};
   return HermesValue::encodeStringValue(
       getRuntime(shr).getStringPrimFromSymbolID(SymbolID::unsafeCreate(symID)));
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_create_private_name(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_create_private_name(
     SHRuntime *shr,
     SHSymbolID descStrID) {
   Runtime &runtime = getRuntime(shr);
@@ -1651,8 +1600,7 @@
   return *cr;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_ljs_create_regexp(SHRuntime *shr, SHSymbolID pattern, SHSymbolID flags) {
   Runtime &runtime = getRuntime(shr);
   CallResult<HermesValue> cr =
@@ -1675,8 +1623,7 @@
   return *cr;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_ljs_create_bigint(SHRuntime *shr, const uint8_t *value, uint32_t size) {
   Runtime &runtime = getRuntime(shr);
   CallResult<HermesValue> res{ExecutionStatus::EXCEPTION};
@@ -1691,8 +1638,7 @@
   return *res;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_new_object(SHRuntime *shr) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_new_object(SHRuntime *shr) {
   PseudoHandle<JSObject> result;
   {
     NoHandleScope noHandle{getRuntime(shr)};
@@ -1700,8 +1646,7 @@
   }
   return result.getHermesValue();
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_new_object_with_parent(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_new_object_with_parent(
     SHRuntime *shr,
     const SHLegacyValue *parent) {
   PseudoHandle<JSObject> result;
@@ -1842,8 +1787,7 @@
   return lv.obj.getHermesValue();
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_new_object_with_buffer(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_new_object_with_buffer(
     SHRuntime *shr,
     SHUnit *unit,
     uint32_t shapeTableIndex,
@@ -1857,8 +1801,7 @@
       valBufferOffset);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_new_object_with_buffer_and_parent(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_new_object_with_buffer_and_parent(
     SHRuntime *shr,
     SHUnit *unit,
     SHLegacyValue *parent,
@@ -1909,8 +1852,7 @@
       runtime, unit, parentHandle, shapeTableIndex, valBufferOffset);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_new_array(SHRuntime *shr, uint32_t size) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_new_array(SHRuntime *shr, uint32_t size) {
   Runtime &runtime = getRuntime(shr);
 
   CallResult<HermesValue> arrayRes = [&runtime, size]() {
@@ -1923,8 +1865,7 @@
   return *arrayRes;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_new_array_with_buffer(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_new_array_with_buffer(
     SHRuntime *shr,
     SHUnit *unit,
     uint32_t numElements,
@@ -1996,8 +1937,7 @@
   return arr;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_ljs_define_own_in_dense_array(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_ljs_define_own_in_dense_array(
     SHRuntime *shr,
     SHLegacyValue *array,
     SHLegacyValue *prop,
@@ -2010,8 +1950,7 @@
       vmcast<JSArray>(*toPHV(array)), runtime, idx, shv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_new_fastarray(SHRuntime *shr, uint32_t sizeHint) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_new_fastarray(SHRuntime *shr, uint32_t sizeHint) {
   Runtime &runtime = getRuntime(shr);
 
   CallResult<HermesValue> arrayRes = [&runtime, sizeHint]() {
@@ -2024,8 +1963,7 @@
   return *arrayRes;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_ljs_is_in_rjs(SHRuntime *shr, SHLegacyValue *name, SHLegacyValue *obj) {
   Runtime &runtime = getRuntime(shr);
   CallResult<bool> cr{false};
@@ -2043,8 +1981,7 @@
     _sh_throw_current(shr);
   return _sh_ljs_bool(*cr);
 }
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_private_is_in_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_private_is_in_rjs(
     SHRuntime *shr,
     SHLegacyValue *privateName,
     SHLegacyValue *target,
@@ -2103,8 +2040,7 @@
   return _sh_ljs_bool(*cr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_pname_list_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_pname_list_rjs(
     SHRuntime *shr,
     SHLegacyValue *base,
     SHLegacyValue *index,
@@ -2151,8 +2087,7 @@
   return *cr;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_get_next_pname_rjs(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_get_next_pname_rjs(
     SHRuntime *shr,
     SHLegacyValue *props,
     SHLegacyValue *base,
@@ -2262,8 +2197,7 @@
   return *result;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue _sh_ljs_direct_eval(
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue _sh_ljs_direct_eval(
     SHRuntime *shr,
     SHLegacyValue *evalText,
     bool strictCaller) {
@@ -2288,13 +2222,11 @@
   return *result;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" int32_t _sh_to_int32_double_slow_path(double d) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE int32_t _sh_to_int32_double_slow_path(double d) {
   return truncateToInt32SlowPath(d);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_prload_direct(SHRuntime *shr, SHLegacyValue source, uint32_t propIndex) {
   Runtime &runtime = getRuntime(shr);
   NoAllocScope noAlloc(runtime);
@@ -2303,8 +2235,7 @@
       .unboxToHV(runtime);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_prload_indirect(SHRuntime *shr, SHLegacyValue source, uint32_t propIndex) {
   Runtime &runtime = getRuntime(shr);
   NoAllocScope noAlloc(runtime);
@@ -2313,8 +2244,7 @@
       .unboxToHV(runtime);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_prstore_direct(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_prstore_direct(
     SHRuntime *shr,
     SHLegacyValue *target,
     uint32_t propIndex,
@@ -2326,8 +2256,7 @@
       vmcast<JSObject>(*toPHV(target)), runtime, propIndex, shv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_prstore_direct_bool(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_prstore_direct_bool(
     SHRuntime *shr,
     SHLegacyValue *target,
     uint32_t propIndex,
@@ -2339,8 +2268,7 @@
       vmcast<JSObject>(*toPHV(target)), runtime, propIndex, shv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_prstore_direct_number(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_prstore_direct_number(
     SHRuntime *shr,
     SHLegacyValue *target,
     uint32_t propIndex,
@@ -2353,8 +2281,7 @@
       vmcast<JSObject>(*toPHV(target)), runtime, propIndex, shv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_prstore_direct_object(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_prstore_direct_object(
     SHRuntime *shr,
     SHLegacyValue *target,
     uint32_t propIndex,
@@ -2367,8 +2294,7 @@
       vmcast<JSObject>(*toPHV(target)), runtime, propIndex, shv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_prstore_direct_string(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_prstore_direct_string(
     SHRuntime *shr,
     SHLegacyValue *target,
     uint32_t propIndex,
@@ -2381,8 +2307,7 @@
       vmcast<JSObject>(*toPHV(target)), runtime, propIndex, shv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_prstore_indirect(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_prstore_indirect(
     SHRuntime *shr,
     SHLegacyValue *target,
     uint32_t propIndex,
@@ -2394,19 +2319,16 @@
       vmcast<JSObject>(*toPHV(target)), runtime, propIndex, shv);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_unreachable() {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_unreachable() {
   hermes_fatal("Unreachable code reached");
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_throw_array_oob(SHRuntime *shr) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_throw_array_oob(SHRuntime *shr) {
   (void)getRuntime(shr).raiseRangeError("array load index out of range");
   _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_fastarray_load(SHRuntime *shr, SHLegacyValue *array, double index) {
   Runtime &runtime = getRuntime(shr);
   auto *arr = vmcast<FastArray>(*toPHV(array));
@@ -2420,8 +2342,7 @@
   return storage->at(intIndex).unboxToHV(runtime);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_fastarray_store(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_fastarray_store(
     SHRuntime *shr,
     const SHLegacyValue *storedValue,
     SHLegacyValue *array,
@@ -2439,8 +2360,7 @@
   return storage->set(intIndex, shv, runtime.getHeap());
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_fastarray_push(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_fastarray_push(
     SHRuntime *shr,
     SHLegacyValue *pushedValue,
     SHLegacyValue *array) {
@@ -2452,8 +2372,7 @@
     _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" void _sh_fastarray_append(
+extern "C" LLVM_ATTRIBUTE_NOINLINE void _sh_fastarray_append(
     SHRuntime *shr,
     SHLegacyValue *other,
     SHLegacyValue *array) {
@@ -2466,8 +2385,7 @@
     _sh_throw_current(shr);
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_string_concat(SHRuntime *shr, uint32_t argCount, ...) {
   Runtime &runtime = getRuntime(shr);
 
@@ -2572,13 +2490,11 @@
   return *result;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" int _sh_errno(void) {
+extern "C" LLVM_ATTRIBUTE_NOINLINE int _sh_errno(void) {
   return errno;
 }
 
-LLVM_ATTRIBUTE_NOINLINE
-extern "C" SHLegacyValue
+extern "C" LLVM_ATTRIBUTE_NOINLINE SHLegacyValue
 _sh_asciiz_to_string(SHRuntime *shr, const char *str, ptrdiff_t len) {
   Runtime &runtime = getRuntime(shr);
   NoHandleScope noHandle{runtime};
